pkg/test/kube/dump.go

```diff
@@ -129,17 +130,35 @@ func DumpPodEvents(a Accessor, workDir, namespace string, pods ...kubeApiCore.Po
 func DumpPodLogs(a Accessor, workDir, namespace string, pods ...kubeApiCore.Pod) {
 	pods = podsOrFetch(a, pods, namespace)
 
+	var sb strings.Builder
 	for _, pod := range pods {
+		isVM := checkIfVM(pod)
 		containers := append(pod.Spec.Containers, pod.Spec.InitContainers...)
 		for _, container := range containers {
 			l, err := a.Logs(pod.Namespace, pod.Name, container.Name, false /* previousLog */)
 			if err != nil {
 				scopes.Framework.Errorf("Unable to get logs for pod/container: %s/%s/%s", pod.Namespace, pod.Name, container.Name)
 				continue
 			}
+			sb.WriteString(l)
+
+			// Get proxy logs if the pod is a VM, since kubectl logs only shows the logs from iptables
+			if isVM && container.Name == "istio-proxy" {
+				if cfgDump, err := a.Exec(pod.Namespace, pod.Name, container.Name, "pilot-agent request GET config_dump"); err == nil {
+					sb.WriteString(cfgDump)
+				} else {
+					scopes.Framework.Errorf("Unable to get istio-proxy config dump for pod: %s/%s", pod.Namespace, pod.Name)
+				}
+
+				if clustersDump, err := a.Exec(pod.Namespace, pod.Name, container.Name, "pilot-agent request GET clusters"); err == nil {
+					sb.WriteString(clustersDump)
+				} else {
+					scopes.Framework.Errorf("Unable to get istio-proxy clusters for pod: %s/%s", pod.Namespace, pod.Name)
+				}
+			}
 
 			fname := path.Join(workDir, fmt.Sprintf("%s-%s.log", pod.Name, container.Name))
-			if err = ioutil.WriteFile(fname, []byte(l), os.ModePerm); err != nil {
+			if err = ioutil.WriteFile(fname, []byte(sb.String()), os.ModePerm); err != nil {
```

Comment:
Might make sense to split this into 3 files, to make it bit more clear

Reply:
Will do! Thanks for pointing it out.